main/src/library/File.flix

```diff
@@ -840,4 +840,29 @@ namespace File {
                 Err(getMessage(ex))
         }
 
+    ///
+    /// Moves a file or directory from path `src` to path `dst`.
+    ///
+    /// Returns `Ok(true)` if `src` was moved, and `dst` did not already exist.
+    /// Returns `Err(msg)` if `src` was not moved, because `dst` already exists or is a subset of `src`.
+    ///
+    pub def move(src: {src :: String} , dst: String): Result[Bool, String] = ???
+
+    ///
+    /// Moves a file or directory from path `src` to directory `dst`.
+    ///
+    /// Returns `Ok(true)` if `src` was moved, and `dst` is a directory.
+    /// Returns `Ok(false)` if `dst` is not a directory.
```

Comment:
Maybe the `false` case makes a bit more sense here, since we need the `dst` to be a directory to `moveInto`.

Reply:
Hmm. I still think Unit here makes the most sense.

Reply:
But I am not 100%

Reply:
We can always change it later